### PR TITLE
e2e: add e2e tests for consul namespaces on ent with acls

### PR DESCRIPTION
**Pull request by shoenig**
~(hold until ent half goes in - making sure oss CI still works)~

This PR cherry-picks changes made for ENT testing of Consul Namespaces with ACLs enabled.

### CODE REVIEW

File: e2e/consul/namespaces_acls_ent.go [hunk 1]
```diff
@@ -0,0 +1,564 @@
+// +build ent
```
Comment (tgross) [inline e2e/consul/namespaces_acls_ent.go:1]: Should this file be here?

**Review by tgross — CHANGES_REQUESTED** [review]
Looks like there's a stray ENT test file here.

File: e2e/consul/namespaces_acls_ent.go (hunk 1 above)
Comment (shoenig) [reply e2e/consul/namespaces_acls_ent.go]: Yikes, nope! It's gone now >_>

**Review by tgross — APPROVED** [review]

File: e2e/consul/namespaces_acls_ent.go (hunk 1 above)
Comment (tgross) [reply e2e/consul/namespaces_acls_ent.go]: Cool. LGTM!

**Pull request merged**